### ISSUE TITLE
SSL/TLS and App Transport Security changes...

### ISSUE THREAD

**Issue by jeffc-kpcc** [SCPR/KPCC-iPhone#268]
Apple has announced a requirement for mandatory App Transport Security (ATS) support, as described here:

https://developer.apple.com/news/?id=12212016b

This is required to properly mandate secure loading of SSL/TLS assets from the network. Apple has pushed this deadline back a bit, but they won't do it forever.

Right now we have a blanket exemption for all domains, which is not ideal. We must move to as many secure resources as possible (ie. KPCC API endpoints) and ensure that any 3rd party services we depend on are also supported.

We can make exemptions on a case-by-case basis, but Apple will probably require an explanation.

**Comment by jgeorges** [comment]
Will there be any additional services we’ll need to purchase/subscribe to in order to meet security requirements?

**Comment by jeffc-kpcc** [comment]
Probably not. My impression (and I could be wrong!) is that we have server-side support for this already, we just need to use it in our app and make sure that the ATS settings confirm it.

A lot of 3rd party services have moved toward SSL/TLS over the past few years, so odds are that the current SDKs and APIs we're using already use it. We just need to define the ATS stuff for it.

**Comment by jeffc-kpcc** [comment]
*As a point of reference to either future-Jeff or another developer...*

We can (and should) leave *both* NSAllowsArbitraryLoadsInWebContent and NSAllowsArbitraryLoads in the app until we support iOS 10 only.

So says an Apple developer (Quinn "The Eskimo", no less!):

[Apple Dev Forums](https://forums.developer.apple.com/thread/66607)

In a nutshell: In iOS 10, NSAllowsArbitraryLoadsInWebContent will allow us to show arbitrary web content in a WKWebView so that Headlines will work, superseding NSAllowsArbitraryLoads. However, in iOS 9 NSAllowsArbitraryLoads will still work as before.

Per Apple's docs, enabling NSAllowsArbitraryLoadsInWebContent will result in further scrutiny, so we should be sure to explain ourselves when submitting the ATS-enabled update ("Setting this key’s value to YES triggers App Store review and requires justification.").

**Comment by jeffc-kpcc** [comment]
We currently have 3 domains that do not support TLS:

**parse.scprdev.org** (used for Parse)
**live.scpr.org** (streaming server)
**media.scpr.org** (static media server)

These will need to be secured before I can resolve this issue.

**Comment by nakedsushi** [comment]
I’ll look into these endpoints for you. Parse.scprdev.org is already being worked on, but stay tuned for the next two.

**Comment by nakedsushi** [comment]
I created tickets for synctree to enable TLS on all 3 so will keep you posted on github when they're ready.

**Comment by nakedsushi** [comment]
https://parse.scprdev.org/ should be good to go.

**Comment by jeffc-kpcc** [comment]
Looks like there is a problem with parse.scprdev.org, on account of a squabble between StartCom (the CA) and pretty much everyone, including Apple. This results in both Mac and iOS web browsers - as well as apps using TLS - to get a revocation error and no longer working under certain circumstances.

From Apple:

https://support.apple.com/en-us/HT202858

"Apple products will block certificates from WoSign and StartCom root CAs if the "Not Before" date is on or after 1 Dec 2016 00:00:00 GMT/UTC."

More info:

https://forum.startcomca.com/viewtopic.php?f=6&t=626
https://en.wikipedia.org/wiki/StartCom#Trustworthiness

This ticket is blocked until this can be resolved.

**Comment by jeffc-kpcc** [comment]
It looks like **parse.scprdev.org** and **live.scpr.org** are both secure now.

Still waiting on **media.scpr.org**.

**Comment by jeffc-kpcc** [comment]
media.scpr.org is now all set, so we should be good to go.
